edit/completion/sorter.go

```diff
@@ -0,0 +1,66 @@
+package completion
+
+import (
+	"errors"
+	"os"
+	"sort"
+
+	"github.com/elves/elvish/eval"
+	"github.com/elves/elvish/eval/vals"
+	"github.com/xiaq/persistent/hashmap"
+)
+
+var (
+	errSorterMustBeFn = errors.New("sorter must be a function")
+)
+
+type rawCandidates []rawCandidate
+
+func (cs rawCandidates) Len() int           { return len(cs) }
+func (cs rawCandidates) Swap(i, j int)      { cs[i], cs[j] = cs[j], cs[i] }
+func (cs rawCandidates) Less(i, j int) bool { return cs[i].text() < cs[j].text() }
+
+func lookupSorter(m hashmap.Map, name string) (eval.Callable, bool) {
+	key := name
+	if !hashmap.HasKey(m, key) {
+		// Use fallback sorter
+		if !hashmap.HasKey(m, "") {
+			return nil, false
+		}
+		key = ""
+	}
+	value, _ := m.Index(key)
+	sorter, ok := value.(eval.Callable)
+	return sorter, ok
+}
+
+func sortRawCandidates(ev *eval.Evaler, customSorter eval.Callable, seed string, rs []rawCandidate) ([]rawCandidate, error) {
+	// default sorter
```

Comment:
Actually, I prefer to remove this special case, but seems it is extremely fast compared to the following.